pycaw/pycaw.py

```diff
@@ -541,10 +541,12 @@ class AudioDevice(object):
     """
     http://stackoverflow.com/a/20982715/185510
     """
-    def __init__(self, id, state, properties):
+    def __init__(self, id, state, properties, dev):
```

Comment:
Just thinking out loud. This would break the interface for people updating right?
I think we probably want to have it optional `dev=None`. Then I don't mind if it breaks later in `EndpointVolume` property when `self._dev` is `None` since it's a new property. But at least it doesn't break in the `AudioDevice` constructor in case it was already used by people directly

Reply:
`AudioDevice` appears to be only constructed in `AudioUtilities.CreateDevice()`.  The additional COM properties and `AudioDeviceState` passed into this constructor makes me think there'd be no reason for one to call this constructor from the "outside".  If this were a language other than Python, then this constructor would be `private` or `internal` :)

If for some reason you're going to the trouble of passing in these COM objects, then you probably are using `AudioUtilities.CreateDevice()`, which has a `dev` parameter already.